eth_tester/backends/pyevm/main.py

```diff
@@ -287,7 +305,33 @@ def __init__(self):
                 "`PyEVMBackend` requires py-evm to be installed and importable. "
                 "Please install the `py-evm` library."
             )
-        self.reset_to_genesis()
+
+        self.account_keys = None  # set below
+        accounts = len(genesis_state) if genesis_state else None
+        self.reset_to_genesis(genesis_parameters, genesis_state, accounts)
+
+    #
+    # Genesis
+    #
+
+    @staticmethod
+    def generate_genesis_params(overrides=None):
+        return get_default_genesis_params(overrides=overrides)
+
+    @staticmethod
+    def generate_genesis_state(overrides=None, accounts=None):
+        account_keys = get_default_account_keys(quantity=accounts)
+        return generate_genesis_state(account_keys=account_keys, overrides=overrides)
+
+    @classmethod
+    def from_genesis_overrides(cls, parameter_overrides=None, state_overrides=None, accounts=None):
```

Comment:
I count three new APIs that are different ways of accomplishing similar things, maybe four if you count the `generate_genesis_state_for_keys` method. That means a lot of new documentation and examples and helping people figure out which one they should use in which situation. Any way we can tighten this up a bit? Ideally there would be one preferred way of setting genesis info.

For example, maybe this method shouldn't exist, and if someone wants to override both parameters and state, then the preferred way would be explicitly, like:

```py
params = PyEVMBackend.generate_genesis_params(overrides=parameter_overrides)
state = PyEVMBackend.generate_genesis_state(overrides=state_overrides, accounts=accounts)
backend = PyEVMBackend(genesis_parameters=params, genesis_state=state)
```

That would be one less method to document, and one less question for where a new user should start when they want to override genesis info. Sure, it's three lines instead of one, but they decompose nicely, into understanding the other public APIs.

Reply:
Personally, I like the simplicity of a one-liner in this situation - It even reduces confusion in some cases, IMO.  I also understand that this does add an opportunity for confusion, and additional maintenance.

Reply:
:heavy_check_mark: